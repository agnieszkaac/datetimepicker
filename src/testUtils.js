export const setDate = (date) => {
  Date.now = jest.fn(() => Date.parse(date));
};

export const setToday = () => {
  const today = "2020-01-01";
  Date.now = jest.fn(() => Date.parse(today));
};
