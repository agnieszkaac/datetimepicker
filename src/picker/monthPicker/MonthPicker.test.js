import React from "react";
import { render } from "@testing-library/react";
import moment from "moment";

import { MonthPicker } from "./index";
import { setToday } from "../../testUtils";

describe("<MonthPicker /> should", () => {
  setToday();

  //default props
  const onPick = jest.fn((_e) => {});
  const props = {
    viewDate: moment(),
    onPick,
  };

  it("match snapshot", () => {
    const { asFragment } = render(<MonthPicker {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("render 12 months", () => {
    const { getAllByTestId } = render(<MonthPicker {...props} />);
    const months = getAllByTestId("month");

    expect(months.length).toBe(12);
  });
});
