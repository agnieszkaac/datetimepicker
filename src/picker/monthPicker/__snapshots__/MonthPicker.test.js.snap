// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MonthPicker /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="month-picker"
  >
    <button
      class="month is-today"
      data-testid="month"
      type="button"
      value="2020-01-01T00:00:00.000Z"
    >
      Jan
    </button>
    <button
      class="month"
      data-testid="month"
      type="button"
      value="2020-02-01T00:00:00.000Z"
    >
      Feb
    </button>
    <button
      class="month"
      data-testid="month"
      type="button"
      value="2020-03-01T00:00:00.000Z"
    >
      Mar
    </button>
    <button
      class="month"
      data-testid="month"
      type="button"
      value="2020-03-31T23:00:00.000Z"
    >
      Apr
    </button>
    <button
      class="month"
      data-testid="month"
      type="button"
      value="2020-04-30T23:00:00.000Z"
    >
      May
    </button>
    <button
      class="month"
      data-testid="month"
      type="button"
      value="2020-05-31T23:00:00.000Z"
    >
      Jun
    </button>
    <button
      class="month"
      data-testid="month"
      type="button"
      value="2020-06-30T23:00:00.000Z"
    >
      Jul
    </button>
    <button
      class="month"
      data-testid="month"
      type="button"
      value="2020-07-31T23:00:00.000Z"
    >
      Aug
    </button>
    <button
      class="month"
      data-testid="month"
      type="button"
      value="2020-08-31T23:00:00.000Z"
    >
      Sep
    </button>
    <button
      class="month"
      data-testid="month"
      type="button"
      value="2020-09-30T23:00:00.000Z"
    >
      Oct
    </button>
    <button
      class="month"
      data-testid="month"
      type="button"
      value="2020-11-01T00:00:00.000Z"
    >
      Nov
    </button>
    <button
      class="month"
      data-testid="month"
      type="button"
      value="2020-12-01T00:00:00.000Z"
    >
      Dec
    </button>
  </div>
</DocumentFragment>
`;
